### PR TITLE
implemented Arange for ragged tensor

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

**Comment by danpovey** [comment]
LGTM.  Could be optimized a bit more, e.g. doing all the transfer to CPU we need in one kernel, but we can leave that for later.

**Comment by qindazhu** [comment]
OK, merging.

RE memory transfer, do you mean use a kenerl to compute those `row_begin` and `row_end` indexes and save to an array (then do one transfer), I suppose we need a loop with `size == ans.NumAxes()` in that kenerl?

**Comment by danpovey** [comment]
yes thats what I meant.  kernel set up is much slower than a single loop
iteration, especially.if.there is cpu syncing.

File: k2/csrc/ragged_ops.cu [hunk 1]
```diff
@@ -1363,6 +1352,47 @@ std::vector<RaggedShape> GetPrefixes(RaggedShape &src,
...
+  int32_t row_begin = begin, row_end = end - 1;
+  for (int32_t cur_axis = axis; cur_axis < num_axes - 1; ++cur_axis) {
+    axes_out[cur_axis - axis].row_splits =
+        axes_in[cur_axis].row_splits.Arange(row_begin, row_end + 1);
+    int32_t row_id = row_begin;
+    // notice here we may do memory copy from GPU to CPU.
+    row_begin = axes_in[cur_axis].row_splits[row_begin];
+    row_end = axes_in[cur_axis].row_splits[row_end];
+    axes_out[cur_axis - axis].row_splits =
+        Minus(axes_out[cur_axis - axis].row_splits, row_begin);
```
Comment (csukuangfj) [inline k2/csrc/ragged_ops.cu:1385]: When `row_begin` is 0, can the call to `Minus` be avoided?

**Comment by csukuangfj** [comment]
> LGTM. Could be optimized a bit more, e.g. doing all the transfer to CPU we need in one kernel, but we can leave that for later.


The same problem exists for `RaggedShape::Index`:
https://github.com/k2-fsa/k2/blob/d5ecd61c2efd9720d5405f381f6ccc5a741fd634/k2/csrc/ragged.cu#L187

**Comment by danpovey** [comment]
OK.  I'm OK to leave that as a TODO or issue for now, unless someone really wants to fix it now.

**Comment by csukuangfj** [comment]
A TODO has been added in #556. I can fix it when LF-MMI is done.

**Pull request merged**